client/src/modules/cohort/index.jsx

```diff
@@ -13,33 +13,52 @@ import {
   createList,
   fetchListsMetaData
 } from 'modules/shopping-list/model/actions';
-import { CohortIcon, EditIcon, ListIcon } from 'assets/images/icons';
+import {
+  ArchiveIcon,
+  CohortIcon,
+  EditIcon,
+  ListIcon
+} from 'assets/images/icons';
 import { getCohortDetails } from './model/selectors';
 import { MessageType } from 'common/constants/enums';
 import { RouterMatchPropType, UserPropType } from 'common/constants/propTypes';
-import ModalForm from 'common/components/ModalForm';
-import { updateCohort } from './model/actions';
+import FormDialog from 'common/components/FormDialog';
+import { archiveCohort, fetchCohortData, updateCohort } from './model/actions';
 import { noOp } from 'common/utils/noOp';
 import DropdownForm from 'common/components/DropdownForm';
 import { getCurrentUser } from 'modules/authorization/model/selectors';
 import PlusIcon from 'assets/images/plus-solid.svg';
+import Dialog from 'common/components/Dialog';
+import ArchivedCohort from 'modules/cohort/components/ArchivedCohort';
 
 class Cohort extends PureComponent {
   state = {
     listFormVisibility: false,
+    showDialog: false,
```

Comment:
in the previous component you've used isDialogVisible, here `showDialog`.
I'm using <prefix>Visibility...maybe we should stay consistent and choose one form?